### PR TITLE
Added support for spec.xrootd.replicas

### PR DESCRIPTION
**Pull request by shivanshs9**
This PR is related to the second task as a pre-requisite for GSoC '20 project.

### CODE REVIEW

File: deploy/crds/qserv.lsst.org_qservs_crd.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spec:
     listKind: QservList
     plural: qservs
     singular: qserv
-  scope: ""
+  scope: Namespaced
```
Comment (fjammes) [inline deploy/crds/qserv.lsst.org_qservs_crd.yaml]: Why adding this field please?

**Comment by fjammes** [comment]
Hi Shivansh,

This is a great start, thanks!

However it is not yet finished. You should also update dynamically xrootd configuration to take in account the new replica. As mentioned in the excerpt of file `configmap/xrootd/etc/xrootd.cf` below, could you please try to implement the TODO feature inside the  qserv-operator template engine (see go code for function `GenerateContainerConfigMap()`)?
```
# TODO Generate xrootd redirector list using go template
set xrootddn = {{.XrootdRedirectorDn}}
all.manager ${xrootddn}-0.${xrootddn}:2131
all.manager ${xrootddn}-1.${xrootddn}:2131
```
Note that above configuration works for 2 replicas only, you have to make it work for `spec.xrootd.replicas` number.

**Comment by shivanshs9** [comment]
I've done changes which I believe would fix the issue. However, I can't get it running... the Xrootd pods crash with the following error in the logs:
`standard_init_linux.go:211: exec user process caused "exec format error"`
I believe it may be due to different qserv image too. Can you please help with that?

**Comment by shivanshs9** [comment]
@fjammes I've fixed the above issue and it works fine now. The issue had been due to using `template.Func` without a correct template name in `template.New`. Now I use the base path (filename) as template name.

> Since the templates created by ParseFiles are named by the base names of the argument files, t should usually have the name of one of the (base) names of the files. **If it does not, depending on t's contents before calling ParseFiles, t.Execute may fail.** In that case use t.ExecuteTemplate to execute a valid template.
(from https://golang.org/pkg/text/template/#Template.ParseFiles)

Here's a snippet of configmap with 4 replicas of xrootd
```
..............
    set xrootddn = qserv-xrootd-redirector
    all.manager ${xrootddn}-0.${xrootddn}:2131
    all.manager ${xrootddn}-1.${xrootddn}:2131
    all.manager ${xrootddn}-2.${xrootddn}:2131
    all.manager ${xrootddn}-3.${xrootddn}:2131

    # - cmsd redirector runs on port 2131
    # - cmsd server does not open server socket
    #   but only client connection to cmsd redirector
    # - xrootd default port is 1094
    if exec cmsd
        xrd.port 2131
    fi

    # Uncomment the following line for detailed xrootd debugging
    # xrootd.trace all debug
kind: ConfigMap
metadata:
  creationTimestamp: "2020-03-11T05:57:44Z"
  labels:
    app: qserv
    app.kubernetes.io/managed-by: qserv-operator
    container: xrootd
    instance: qserv
  name: qserv-xrootd-etc
  namespace: default
...............
```

**Review by fjammes — COMMENTED** [review]
Code has been integration in DM-24332, accordingly to project development policies.

File: deploy/crds/qserv.lsst.org_qservs_crd.yaml (hunk 1 above)
Comment (shivanshs9) [reply deploy/crds/qserv.lsst.org_qservs_crd.yaml]: For some reason, it was added by the Vscode extension while saving. :sweat_smile:

**Pull request closed without merging**